### PR TITLE
[ETL-574] Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by philerooski**
Things to look out for:

- incorrect information
- incomplete information
- additional documentation we can link out to, or information which is redundant with documentation elsewhere.
- broken links

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+## Jira
+
+Before there was work, there was the _work request_. Work requests a.k.a. tickets for this repository are tracked in the [ETL Jira project](https://sagebionetworks.jira.com/jira/software/c/projects/ETL/boards/190). No development work should be done without first having a Jira ticket which outlines the context and goals of the work to be done.
+
+## Developing and testing locally
+
+1. Clone this repository (do not fork). Create a feature branch off the `main` branch. Branch names must contain the identifier of the corresponding Jira ticket (e.g., `etl-574`). Optionally, add keywords describing the work that is being done to the branch name (e.g., `etl-574-add-contributing-doc`).
+
+> [!IMPORTANT]
+> Keep branch names as short as possible. We use the branch name to namespace resources in AWS (as described below). Since AWS imposes limits on the name length of some resources, the deployment may fail if the branch name is too long.
```
Comment (BryanFauble) [inline CONTRIBUTING.md:31]: IIRC capital letters in the branch name also causes issues

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+2. We use [pipenv](https://pipenv.pypa.io/en/latest/) to manage our development environment. Simply run
+
+```
+pipenv shell
+pipenv install --dev
+```
+
+from the repository root to create a virtual environment and install all dependencies.
```
Comment (BryanFauble) [inline CONTRIBUTING.md:40]: The formatting ended up a little poor here when looking at the rendered md - Ideally this should maintain the same indent level:
image

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+3. Make any number of commits as makes sense to you. These commits may need to be consolidated later on during code review.
```
Comment (BryanFauble) [inline CONTRIBUTING.md:42]:
```suggestion
3. Make any number of commits that solve the problem or leave the project in a better place. These commits may need to be consolidated later on during code review.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+To submit SQS events directly to the [S3 to Glue Lambda](src/lambda_function/s3_to_glue), follow the instructions in its README.
+
+### Submitting test events to the S3 to JSON Glue workflow
+
+To run the S3 to JSON Glue workflow directly, either edit the workflow by adding a `--messages` workflow argument which matches the format of the S3 to Glue Lambda, or start a workflow run and add the `--messages` argument programmatically (mimicking the behavior of the S3 to Glue Lambda).
+
+### JSON to Parquet Workflow
+
+Once the S3 to JSON workflow has successfully complete, the JSON to Parquet workflow will need to be run manually to produce the Parquet datasets, which are written to a namespaced location in `s3://recover-dev-processed-data`. This workflow does not require any configuration once deployed and can be run directly without passing any additional workflow run arguments.
```
Comment (BryanFauble) [inline CONTRIBUTING.md:68]:
```suggestion
Once the S3 to JSON workflow has successfully completed, the JSON to Parquet workflow will need to be run manually to produce the Parquet datasets, which are written to a namespaced location in `s3://recover-dev-processed-data`. This workflow does not require any configuration once deployed and can be run directly without passing any additional workflow run arguments.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+#### Record count comparion
+
+We use a structured logging framework in our Glue jobs so that logs can be consumed in a programmatic way. At the time of writing, this is limited to [a script](src/scripts/consume_logs/consume_logs.py) that will compare the count of lines read versus the count of lines written for each NDJSON file in each S3 to JSON workflow run within a given time frame.
+
+## Code review
+
+Once integration testing is complete,  submit a pull request against the `main` branch. At least one approving review is required to merge, although it is common courtesy to give everyone on the [review team](https://github.com/orgs/Sage-Bionetworks/teams/recover) time to approve the PR.
+
+Before and/or after code review, clean up your commit history. If the `main` branch has changed since you last pushed your branch, [rebase](https://git-scm.com/docs/git-rebase) on main. If you have multiple commits, make sure that each commit is focused on delivering one complete feature. If you need to consolidate commits, consider doing an interactive rebase.
```
Comment (BryanFauble) [inline CONTRIBUTING.md:86]: Maybe we should mention squashing commits as well?

**Review by BryanFauble — APPROVED** [review]
Changes LGTM! Great work on the documentation

**Review by thomasyu888 — APPROVED** [review]
🔥 Fantastic!  I just added some comments - but pre-approving!

File: CONTRIBUTING.md (hunk 1 above)
Comment (philerooski) [reply CONTRIBUTING.md]: I couldn't find a good reference for safe characters in AWS resource names, so I added a generic disclaimer about using safe characters ([a-z], hyphen, underscore)

File: CONTRIBUTING.md (hunk 2 above)
Comment (philerooski) [reply CONTRIBUTING.md]: I don't know if it's possible to put code blocks at an indent level, so I moved the code block to the bottom of point 2 so the formatting is less jarring.

**Pull request merged**